system/jlib/jptree.cpp

```diff
@@ -1305,11 +1305,11 @@ bool PTree::isCompressed(const char *xpath) const
     else
     {
         const char *prop = splitXPathX(xpath);
-        if (prop && '\0' != *prop && !isAttribute(prop))
+        if (!isAttribute(prop))
         {
             IPropertyTree *branch = queryPropTree(xpath);
             if (branch)
-                return branch->isCompressed(prop);
```

Comment:
Was passing in prop here wrong ?

Reply:
Yes. splitXPathX returns a pointer to the tail property, so that it can see if it's an attribute or not (if attribute then it cannot be compressed/binary), splitXPathX does not alter the input param (xpath).

The queryPropTree(xpath) that is unchanged here, gets to the property required, so the bug was to then ask that property if it had the tail property in it.

e.g. isCompressed("a/b"), prior to this fix, this code would incorrectly do:
```
branch = queryPropTree("a/b")
branch->isCompressed("b");
```